### PR TITLE
Storing last zip used if it's valid in SharedPreferences

### PR DESCRIPTION
**Pull request by chelseybaker**
[Trello card](https://trello.com/c/N0gp38y6/6-app-should-remember-the-last-location-typed-in)

Stores the last used zip code and repopulates it!

### CODE REVIEW

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java [hunk 1]
```diff
@@ -198,4 +205,19 @@ private void printGSON(final Response<TenDayForecast> response) {
...
     }
+
+    private void saveZip(String zip) {
+        SharedPreferences.Editor editor = getPreferences(0).edit();
+        editor.putString(StoredZipKey, zip);
+        editor.commit();
+        Log.d("LandingActivity", "Saved zip");
+    }
+
+    private void setSavedZip() {
```
Comment (stkent) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: This method name caught me out at first; I thought _it_ was the one that would save the ZIP code! Maybe something like "loadSavedZip" or "populateSavedZip" might be clearer?

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java [hunk 2]
```diff
@@ -198,4 +205,19 @@ private void printGSON(final Response<TenDayForecast> response) {
         final Gson gson = new GsonBuilder().setPrettyPrinting().create();
         Log.d("LandingActivity", gson.toJson(response.body()));
     }
+
+    private void saveZip(String zip) {
+        SharedPreferences.Editor editor = getPreferences(0).edit();
```
Comment (stkent) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: When calling `getPreferences`, even though the method signature "only" requires an `int`, we're actually expected to pass one out of a finite collection of "allowed" constants. The value 0 is valid for this method and actually corresponds to the "allowed" constant [`MODE_PRIVATE`](https://developer.android.com/reference/android/content/Context.html#MODE_PRIVATE). Usually documentation for a method with an ambiguous signature like this one will indicate where you can find the list of allowed values.

Bottom line: we should update to use `MODE_PRIVATE` here to be more correct/communicative!

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java [hunk 3]
```diff
@@ -105,6 +110,8 @@ public void onClick(final View v) {
             return;
         }
 
+        saveZip(zipCode);
```
Comment (stkent) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: In production I'd probably avoid saving this until we know the ZIP is valid (i.e. the network request completes successfully). Not something I'm going to hold this PR up for though; just noting :)

**Comment by stkent** [comment]
@chelseybaker comments in!

**Comment by stkent** [comment]
ur a zipsaver!

bitmoji

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java (hunk 2 above)
Comment (chelseybaker) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: I stole the 0 from the Android DOCS! Shame on them :P https://developer.android.com/guide/topics/data/data-storage.html#pref

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java (hunk 3 above)
Comment (chelseybaker) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: Validation is in the next card!
I did also purposely place this after my two mini validation checks though. But yes, working on validation right meow

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java (hunk 2 above)
Comment (stkent) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: Those bastards!

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java (hunk 3 above)
Comment (stkent) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: 👍 awesome!

File: app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java (hunk 1 above)
Comment (chelseybaker) [inline app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java]: Good call, renamed. 

And side note, I was renaming it manually, then went back and undid that all and used the refactor method. We don't have that in the land of iOS still so a good ol search and replace rename is what I'm still used to 😭

**Comment by stkent** [comment]
bitmoji
